api/user.go

```diff
@@ -1621,33 +1622,42 @@ func resetPassword(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	userId := props["user_id"]
-	hash := props["hash"]
-	timeStr := ""
 
-	if !c.IsSystemAdmin() {
-		if len(hash) == 0 {
-			c.SetInvalidParam("resetPassword", "hash")
+	if !c.IsSystemAdmin() || len(userId) == 0 {
```

Comment:
Maybe the system admins ability to reset passwords should be a separate route in the admin routes. Then they can share a helper function that actually does the reset. This function is a bit confusing. Normally not important but it is a critical security function.